src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -5,46 +5,69 @@
 {
     internal class TemplateCache
     {
+        private const string BulletSymbol = "\u2022";
+
         private readonly ILogger _logger;
 
-        public TemplateCache(ScanResult?[] scanResults, Dictionary<string, DateTime> mountPoints, ILogger logger)
+        public TemplateCache(IReadOnlyList<ITemplatePackage> allTemplatePackages, ScanResult?[] scanResults, Dictionary<string, DateTime> mountPoints, ILogger logger)
         {
             _logger = logger;
+
             // We need this dictionary to de-duplicate templates that have same identity
             // notice that IEnumerable<ScanResult> that we get in is order by priority which means
-            // last template with same Identity wins, others are ignored...
-            var templateDeduplicationDictionary = new Dictionary<string, (ITemplate Template, ILocalizationLocator? Localization)>();
+            // last template with same Identity will win, others will be ignored...
+            var templateDeduplicationDictionary = new Dictionary<string, IList<(ITemplate Template, string PackageDisplayName, ILocalizationLocator? Localization)>>();
             foreach (var scanResult in scanResults)
             {
                 if (scanResult == null)
                 {
                     continue;
                 }
+
                 foreach (ITemplate template in scanResult.Templates)
                 {
-                    templateDeduplicationDictionary[template.Identity] = (template, GetBestLocalizationLocatorMatch(scanResult.Localizations, template.Identity));
+                    var templatePackageDisplayName = allTemplatePackages.FirstOrDefault(tp => tp.MountPointUri == template.MountPointUri) is IManagedTemplatePackage managedTP
+                        ? managedTP.DisplayName
+                        : string.Empty;
+
+                    if (templateDeduplicationDictionary.ContainsKey(template.Identity))
+                    {
+                        templateDeduplicationDictionary[template.Identity].Add((template, templatePackageDisplayName, GetBestLocalizationLocatorMatch(scanResult.Localizations, template.Identity)));
+                    }
+                    else
+                    {
+                        templateDeduplicationDictionary[template.Identity] = new List<(ITemplate Template, string PackageDisplayName, ILocalizationLocator? Localization)>
+                        {
+                            (template, templatePackageDisplayName, GetBestLocalizationLocatorMatch(scanResult.Localizations, template.Identity))
+                        };
+                    }
                 }
             }
 
             var templates = new List<TemplateInfo>();
-            foreach (var newTemplate in templateDeduplicationDictionary.Values)
+            foreach (var duplicatedIdentities in templateDeduplicationDictionary)
             {
+                var newTemplate = duplicatedIdentities.Value.LastOrDefault();
```

Comment:
it would be good to include the comment why the last is taken, also `OrDefault` is not needed.